src/cmd/sign.js

```diff
@@ -111,38 +131,64 @@ export default function sign(
         log.warn('No extension ID specified (it will be auto-generated)');
       }
 
-      const signingResult = await signAddon({
+      if (useSubmissionApi && !channel) {
+        throw new UsageError(
+          'channel is a required parameter for the addon submission API'
+        );
+      }
+
+      if (useSubmissionApi && apiProxy) {
+        // https://github.com/mozilla/web-ext/issues/2472
+        throw new UsageError(
+          "apiProxy isn't yet supported for the addon submission API"
+        );
+      }
+
+      if (useSubmissionApi && apiHost.endsWith('/')) {
+        apiHost = apiHost.slice(0, -1);
+      }
+
+      const signSubmitArgs = {
         apiKey,
         apiSecret,
-        apiUrlPrefix,
-        apiProxy,
         timeout,
-        verbose,
         id,
         xpiPath: buildResult.extensionPath,
-        version: manifestData.version,
         downloadDir: artifactsDir,
         channel,
-      });
-
-      if (signingResult.id) {
-        await saveIdToSourceDir(sourceDir, signingResult.id);
-      }
-
-      // All information about the downloaded files would have
-      // already been logged by signAddon().
-      if (signingResult.success) {
-        log.info(`Extension ID: ${signingResult.id}`);
-        log.info('SUCCESS');
-      } else {
+      };
+
+      let result;
+      try {
+        if (useSubmissionApi && channel) {
+          // we verify `channel` is set above, this is just for Flow
```

Comment:
maybe use something like `// $FlowIgnore: we verify 'channel' is set above` then?

Reply:
i.e. don't have the `&& channel` in the `if`?  (`if (useSubmissionApi) {`) and just ignore the Flow error instead?

Reply:
Yes